cmd/secretless-broker/main.go

```diff
@@ -2,12 +2,48 @@ package main
 
 import (
 	"flag"
+	"os"
+	"os/signal"
+	"syscall"
 
-	"github.com/cyberark/secretless-broker/internal/plugin"
 	"github.com/cyberark/secretless-broker/pkg/secretless"
+	v2 "github.com/cyberark/secretless-broker/pkg/secretless/config/v2"
+	"github.com/pkg/profile"
 )
 
 func main() {
+	params := CmdLineParams()
+
+	// TODO: pass in logger, config, eventNotifier, what else?
+
+	var test []*v2.Service
+
+	// checksum logic
+
+	// allAvailablePlugins
+
+	// validate requested plugins
+
+	cfg := v2.Config{
+		Debug: false,
+		Services: test,
+	}
+
+	secretless := NewSecretless()
+	perfProfile := NewPerformanceProfile()
+
+	StopOnKillSignal(secretless)
```

Comment:
`StopOnKillSignal`: `kill` is only one of the signals that we watch for. maybe we should change the name to not be specific to signal type.

Reply:
yeah good point i was thinking of "kill" as generic but you're right it's also a specific signal.  any suggestions?  `StopOnAnyHaltSignal` is verbose but perhaps clear?

Reply:
`StopOnProgramExit`? 🤷‍♂